querydsl-jpa/src/main/java/com/querydsl/jpa/hibernate/AbstractHibernateQuery.java

```diff
@@ -153,7 +166,7 @@ private Query createQuery(@Nullable QueryModifiers modifiers, boolean forCount)
   public CloseableIterator<T> iterate() {
     try {
       Query query = createQuery();
-      ScrollableResults results = query.scroll(ScrollMode.FORWARD_ONLY);
+      List results = query.getResultList();
       return new ScrollableResultsIterator<T>(results);
```

Comment:
Do you think something like this would work instead?

```
        Stream<T> stream = stream();
        return new IteratorAdapter<T>(stream.iterator(), stream::close);
```

Reply:
Unfortunately this doesn't work. Under the hood `stream()` is using the same iterator that is causing issues in the first place.